src/main/java/frc/robot/subsystems/IndexSubsystem.java

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.util.Color;
+
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.ColorMatchResult;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+import com.revrobotics.ColorMatch;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix.motorcontrol.TalonSRXControlMode;
+import frc.robot.Constants;
+
+
+public class IndexSubsystem extends SubsystemBase {
+
+  // Should this be in constants?
+  private final I2C.Port i2cPort = I2C.Port.kOnboard;
+  int i2cPortId = 0x70;
+  public final int ColorSensorid = 0;
+
+  // Senses colors
+  private final ColorSensorV3 colorSensor = new ColorSensorV3(i2cPort);
+  // Matches colors
+  private final ColorMatch colorMatcher = new ColorMatch();
+
+  private final Color kBlueTarget = new Color(0.143, 0.427, 0.429);
+  private final Color kRedTarget = new Color(0.562, 0.351, 0.100);
+  private final Color kBlackTarget = new Color(0.252, 0.483, 0.263);
+  private Color detectedColor;
+
+
+
+
+  // Motors
+  private TalonSRX front = new TalonSRX(Constants.FRONT_INDEX_ID);
+  private TalonSRX tower = new TalonSRX(Constants.TOWER_INDEX_ID);
+  private TalonSRX back = new TalonSRX(Constants.BACK_INDEX_ID);
+  
+  public static void MuxChangeI2cPort(I2C.Port I2CPort,int deviceAdress,int newPort) {
+  I2C I2CObject = new I2C(I2CPort,deviceAdress);
+  int i2cPortId = 0x70; // MUX I2C address
+  // and you simply write a single byte with the desired multiplexed output
+  // number to that port
+   boolean failed = I2CObject.write(i2cPortId, newPort);
+   if (failed) {
+      throw new RuntimeException("Failed to write to MUX over I2C");
```

Comment:
`RuntimeException`s should **never** be used. A subsystem failing to operate as intended is okay, but killing the entire robot mid-match because a color sensor can't be reached should never happen.